```
❯ go test -race k8s.io/apiserver/pkg/registry/generic/registry -c

❯ stress ./registry.test -test.run=TestUnsafeDeleteWithAdmissionShouldBeSkipped
5s: 22 runs so far, 0 failures
10s: 53 runs so far, 1 failures (1.89%)
15s: 82 runs so far, 1 failures (1.22%)
20s: 111 runs so far, 1 failures (0.90%)
25s: 142 runs so far, 1 failures (0.70%)
...
5m20s: 1864 runs so far, 4 failures (0.21%)
5m25s: 1894 runs so far, 4 failures (0.21%)
5m30s: 1924 runs so far, 4 failures (0.21%)
5m35s: 1954 runs so far, 4 failures (0.20%)
5m40s: 1983 runs so far, 4 failures (0.20%)
```

The 4 failures attribute to the following error:
```
/var/folders/0c/v7fg59_s2qn6kllppph7g1kh0000gn/T/go-stress-20250307T135026-2203497921
--- FAIL: TestUnsafeDeleteWithAdmissionShouldBeSkipped (0.00s)
    logger.go:146: 2025-03-07T13:53:28.531-0500	ERROR	etcd-server	creating peer listener failed	{"error": "listen tcp 127.0.0.1:53102: bind: address already in use"}
    test_server.go:40: listen tcp 127.0.0.1:53102: bind: address already in use
FAIL
```

It points to: https://github.com/kubernetes/kubernetes/blob/a6ad9aa8ada1866e7603c93d0e09b2e0e96826b3/staging/src/k8s.io/apiserver/pkg/storage/etcd3/testserver/test_server.go#L37-L49

--- a/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/corrupt_obj_deleter.go
+++ b/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/corrupt_obj_deleter.go
@@ -107,8 +107,14 @@ func (d *corruptObjectDeleter) Delete(ctx context.Context, name string, deleteVa
 	klog.FromContext(ctx).V(1).Info("Going to perform unsafe object deletion", "object", klog.KRef(genericapirequest.NamespaceValue(ctx), name))
 	out := d.store.NewFunc()
 	storageOpts := storage.DeleteOptions{IgnoreStoreReadError: true}
-	// dropping preconditions, and keeping the admission
-	if err := storageBackend.Delete(ctx, key, out, nil, storage.ValidateObjectFunc(deleteValidation), nil, storageOpts); err != nil {
+	// we don't have the old object in the cache, neither can it be
+	// retrieved from the storage and decoded into an object
+	// successfully, so we do the following:
+	//  a) skip preconditions check
+	//  b) skip admission validation, rest.ValidateAllObjectFunc will "admit everything"
+	var nilPreconditions *storage.Preconditions = nil
+	var nilCachedExistingObject runtime.Object = nil
+	if err := storageBackend.Delete(ctx, key, out, nilPreconditions, rest.ValidateAllObjectFunc, nilCachedExistingObject, storageOpts); err != nil {
 		if storage.IsNotFound(err) {
 			// the DELETE succeeded, but we don't have the object since it's
 			// not retrievable from the storage, so we send a nil object

--- a/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/corrupt_obj_deleter_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/corrupt_obj_deleter_test.go
@@ -267,6 +267,50 @@ func TestUnsafeDeleteWithUnexpectedError(t *testing.T) {
 	}
 }
 
+func TestUnsafeDeleteWithAdmissionShouldBeSkipped(t *testing.T) {
+	ctx := genericapirequest.WithNamespace(genericapirequest.NewContext(), "test")
+	destroyFunc, registry := NewTestGenericStoreRegistry(t)
+	defer destroyFunc()
+
+	// a) create the target object
+	object := &example.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+		Spec:       example.PodSpec{NodeName: "machine"},
+	}
+	_, err := registry.Create(ctx, object, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// b) wrap the storage layer to return corrupt object error
+	registry.Storage.Storage = &corruptStorage{
+		Interface: registry.Storage.Storage,
+		err:       storage.NewCorruptObjError("key", fmt.Errorf("untransformable")),
+	}
+
+	// c) set up a corrupt object deleter for the registry
+	deleter := NewCorruptObjectDeleter(registry)
+
+	// d) try unsafe delete, but pass a validation that always fails
+	var admissionInvoked int
+	_, deleted, err := deleter.Delete(ctx, object.Name, func(_ context.Context, _ runtime.Object) error {
+		admissionInvoked++
+		return fmt.Errorf("admission was not skipped")
+	}, &metav1.DeleteOptions{
+		IgnoreStoreReadErrorWithClusterBreakingPotential: ptr.To[bool](true),
+	})
+
+	if err != nil {
+		t.Errorf("Unexpected error from Delete: %v", err)
+	}
+	if want, got := true, deleted; want != got {
+		t.Errorf("Expected deleted: %t, but got: %t", want, got)
+	}
+	if want, got := 0, admissionInvoked; want != got {
+		t.Errorf("Expected admission to be invoked %d time(s), but got: %d", want, got)
+	}
+}
+
 type corruptStorage struct {
 	storage.Interface
 	err                 error

